File: examples/fsm_platformer/fsm_platformer_unity/FallTransition.cs
using UnityEngine;

// Unity APIs used: Rigidbody2D.linearVelocity
public class FallTransition : FSMTransition
{
    private PlayerInputHandler player;

    public FallTransition(FSMState targetState, PlayerInputHandler player) : base(targetState)
    {
        this.player = player;
    }

    public override bool IsValid(FSMState currentState)
    {
        return player.Rb.linearVelocity.y <= 0;
    }
}

File: examples/fsm_platformer/fsm_platformer_python/fall_transition.py
"""Fall Transition — JUMPING -> FALLING when vertical velocity <= 0."""

from __future__ import annotations

from fsm_platformer_python.fsm import FSMState, FSMTransition


class FallTransition(FSMTransition):
    """Fires when the player's vertical velocity becomes non-positive (apex of jump)."""

    def __init__(self, target_state: FSMState, player_input_handler) -> None:
        super().__init__(target_state)
        self.player = player_input_handler

    def is_valid(self, current_state: FSMState) -> bool:
        return self.player.rb.velocity.y <= 0
